### ISSUE TITLE
Possible unexpected behaviour with trigger: 'click mouseenter'

### ISSUE THREAD

**Issue by mattsbennett** [atomiks/tippyjs#649]
## Bug description

As in the title, I _think_ there might be unexpected behaviour with `trigger: 'click mouseenter'`. It also could be that I have expectations that don't match the intent of the existing option interactions!

## Reproduction

Very simple case: https://codepen.io/mattsbennett/pen/XWJJNqr

Here's what I would expect:

- Action: Hover element.
    - Result: Tip shown (expected)
- Action: Element still hovered. Now, click tip.
    - Result: Tip hidden (unexpected)
    - Expected Result: Tip remains shown
- Action: Element still hovered. Click tip again.
    - Result: Tip shown (expected)
- Action: Move cursor so element no longer hovered.
    - Result: Tip hidden (unexpected)
    - Expected result: Tip shown until clicking outside element (as per default `hideOnClick` and `trigger: 'click'`).

In other words, if `click` _and_ `mouseenter` are specified in `trigger` the tip should (I think) remain shown on clicking the element and until the element is clicked outside (as per `hideOnClick`) (instead of the existing toggle-on-click and always-hide-on-mouseleave behaviour).

So, `mouseenter` and `mouseleave` would toggle the tip until the element is clicked, after which only clicking outside the element would close it (assuming the default `hideOnClick: true`). I believe this would be more consistent with the behaviour of `trigger: 'click'` on its own, and still respect the behaviour of `trigger: 'mouseenter'`.

**Comment by atomiks** [comment]
That kind of behavior is somewhat ambiguous. I don't think it makes sense to have `mouseenter` and `click` in the same trigger in the first place? 🤔 What is the use case for it?

**Comment by mattsbennett** [comment]
Gotcha - I didn't realize the intent was that these two triggers would not be used together. If that's the case, perhaps a clarification in the docs might help? That said, here's my use-case:

For interactive tooltips/popovers that contain buttons/other interactive UI elements, having to hover the trigger element, then carefully guide the cursor over top of the tip to get to the other elements is less than desirable UX. The proposed behaviour above would provide the following benefits:

- `mouseenter` would aid in discoverability of the interactive tip (currently `click` triggered tips are not very discoverable for use in any common user-flow).
- Having the `click` toggle the tip open until clicked outside would resolve the cursor-guiding requirement mentioned above.

**Comment by atomiks** [comment]
Changing the current behavior would be breaking for anyone relying on it :-( Maybe there is a way to do what you're looking for by dynamically changing the `trigger` via `.setProps` in the lifecycles.

**Comment by mattsbennett** [comment]
The current behaviour: Tip shows on hovering the target and clicking the target toggles it - but only while hovered. I don't see how this provides any advantage for usability or otherwise, whereas the proposed behaviour does. So, though it would change the current behaviour, I'd suggest it is would be an enhancement.

That said, just my 2 cents of course! Clearly your call as the owner/maintainer.

In case you, or anyone else is interested, I created a plugin that provides this behaviour (probably could be improved, but works for now):

```
const pinOnClick = {
    name: "pinOnClick",
    defaultValue: true,
    fn(instance) {
        const handleClick = (e) => {
            e.preventDefault();
            instance.setProps({
                hideOnClick: true,
                trigger: "click",
            });
            instance.show();
        };
        const originalProps = {
            hideOnClick: instance.props.hideOnClick,
            trigger: instance.props.trigger,
        };

        return {
            onCreate() {
                if (
                    instance.props.trigger.includes("click") &&
                    instance.props.trigger.includes("mouseenter")
                ) {
                    instance.reference.addEventListener("click", handleClick);
                }
            },
            onHide() {
                instance.setProps(originalProps);
            },
        };
    },
};
```
Finally, thank you for sharing `tippy.js` and for your dialogue on this!
